### PR TITLE
Fix pip install and add more exhaustive CI checks

### LINKED ISSUE #594: Missing dependencies after pip install
Mirrors https://github.com/GlacioHack/xdem/issues/557, fiona is now only an optional dependency of geopandas, so our code breaks

### PR DESCRIPTION
**Pull request by rhugonnet**
This PR removes `fiona` as a dependency of GeoUtils as it was also dropped by GeoPandas in favor of `pyogrio` (which was breaking the `pip` install, and probably would have eventually broken the `conda` one as well).

This PR also adds a new (fast) CI action that simply checks that the `pip install` works and can import the package (usually runs in less than 2min).

Resolves #594

### CODE REVIEW

File: tests/test_misc.py [hunk 1]
```diff
@@ -118,8 +118,8 @@ def useless_func() -> int:
 
     def test_diff_environment_yml(self, capsys) -> None:  # type: ignore
         # Test with synthetic environment
-        env = {"dependencies": ["python==3.9", "numpy", "fiona"]}
-        devenv = {"dependencies": ["python==3.9", "numpy", "fiona", "opencv"]}
+        env = {"dependencies": ["python==3.9", "numpy", "pandas"]}
```
Comment (adehecq) [inline tests/test_misc.py:121]: why did you add pandas instead here?

Comment (rhugonnet) [reply tests/test_misc.py:121]: It is just a synthetic list of dependencies for running the tests (so it would still have worked without this change), but I didn't want to leave "fiona" in there to avoid confusion, as it's not a dependency to our packages anymore, so I simply modified the synthetic list using "pandas" instead.
